File: src/packets/PacketC2SDeleteZone.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x05
 * Delete Zone
 * Deletes a Zone
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unsigned Char) Zone ID
 */
class PacketC2SDeleteZone extends PacketC2S {
    parseData() {
        this.ctrllrID = this._buffer.readUInt8();
        this.zoneID = this._buffer.readUInt8();
    }
}

module.exports = PacketC2SDeleteZone;

File: src/RNetPi.Core/Packets/PacketC2SDeleteZone.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x05
/// Delete Zone
/// Deletes a Zone
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
/// </summary>
public class PacketC2SDeleteZone : PacketC2S
{
    public const byte ID = 0x05;

    public byte ControllerID { get; private set; }
    public byte ZoneID { get; private set; }

    public PacketC2SDeleteZone(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        ControllerID = Reader.ReadByte();
        ZoneID = Reader.ReadByte();
    }
}